### PR TITLE
T002 - Refactored export functionality to improve maintainability and separation of concerns

### PR DESCRIPTION
**Pull request by LizzzzHFFF**
Summary
- Added TrvObservationSummary model for TRV_OBSERVATION_SUMMARY
- Enhanced export to include complete observation summaries
- Restore legacy field export columns
- Add turtle status and measurements export
- Restore flipper and PIT tag export fields
- Align CSV and XLSX field export headers
- Extract export helper functions from views

Changes:
- Moved export configuration (headers, field mappings, constants) out of views.py into export_config.py.
- Moved export processing and helper functions out of views.py into export_helpers.py.
- Kept views.py focused on request handling and export orchestration.
- No intended functional changes to export outputs.
- Verified export results against existing example files to ensure output consistency.
-  Align field and processed observation export with agreed format

Added export fields
- left_flipper_tags,
- right_flipper_tags,
- unknown_flipper_tags,
- all_flipper_tags,
- left_pit_tags,
- right_pit_tags,
- unknown_pit_tags,
- all_pit_tags,
- all_damage
- all_measurements
- all_samples
- observation_status

Added Alive Status filter to Export Data page
- Added Alive Status lookup endpoint
- Added Alive Status filtering for Field and Processed exports

Compatibility
- Retained existing export fields:
  - measurement_1_*
  - measurement_2_*
  - damage_1_*
  - damage_2_*
- Available statuses:1. Yes 2. No 3. Possible 4. Uncertain 5. Didn't Check 6. NA

Testing
- Verified options populate correctly
- Verified export filtering works for all status values
- Verified existing export filters continue to function
- Verified CSV export output
- Verified XLSX export output

### CODE REVIEW

**Review by LizzzzHFFF — COMMENTED** [review]
All done, and check the orders in the export files.

**Review by RickWangPerth — COMMENTED** [review]
Please add focused unit tests for CSV/XLSX header order and populated values, `ENTRY_ID`, empty and multi-related records, organisation scoping, and exports above SQL Server's parameter limit.

Code quality: `git diff --check` currently reports trailing whitespace and a final blank line. Please also remove duplicate imports and commented-out legacy code, and keep the export row-building path single-sourced and tested.

File: wamtram2/export_helpers.py [hunk 1]
```diff
@@ -0,0 +1,1173 @@
...
+
+    "DATE_ENTERED": (
+        _attr(observation, "date_entered").date()
+        if _attr(observation, "date_entered")
+        else ""
+    ),
+    "OBSERVATION_DATE_OLD": _attr(observation, "observation_date_old"),
+    "ALIVE": _raw_fk(observation, "alive"),
+
+    "ENTRY_ID": "",
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py:612]: **P1: `ENTRY_ID` is exported as an empty value for every processed row.**

The associated `TrtDataEntry` is already available as `de`, and `DATA_ENTRY_ID` is populated immediately below. Please populate `ENTRY_ID` from the agreed legacy source (likely `de.data_entry_id`) or remove the column if it is intentionally unavailable. A silently blank required column makes the export incomplete.

File: wamtram2/export_helpers.py [hunk 2]
```diff
@@ -0,0 +1,1173 @@
...
+    ):
+        context["identifications"][
+            identification.observation_id
+        ].append(identification)
+
+    return context
+
+def get_processed_export_row(entry, context):
+
+    observation = entry
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py:511]: **P1: The detailed observations fetched into `context["observations"]` are never used.**

`get_processed_export_row()` assigns `observation = entry`, so later related-object access falls back to per-row lazy queries despite the prefetch work above. Please resolve the observation from `context["observations"]` with a safe fallback, or put the complete `select_related(...)` on the export queryset.

File: wamtram2/export_helpers.py [hunk 3]
```diff
@@ -0,0 +1,1173 @@
...
+    }
+
+    if not observation_ids:
+        return context
+
+    context["observations"] = {
+        observation.observation_id: observation
+        for observation in _safe_queryset(
+            TrtObservations.objects.filter(
+                observation_id__in=observation_ids
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py:386]: **P1: Large exports can silently lose their related data on SQL Server.**

The complete observation-id set is passed through several `__in` queries in this helper. Once it exceeds SQL Server's parameter limit, `_safe_queryset()` catches the `DatabaseError` and returns `[]`, so the download can succeed with related columns blank. Please query IDs in safe chunks and surface database failures rather than converting them to empty export data.

File: wamtram2/views.py [hunk 4]
```diff
@@ -2343,283 +2408,240 @@ def export_data(self, request):
...
+                )
+            )
 
-                # 2. PIT Tags
-                recorded_pit_tags = (
-                    TrtRecordedPitTags.objects.filter(observation_id__in=obs_ids).select_related("pittag_id").order_by("pit_tag_position")
+            summary_dict = {
+                s.observation_id: s
+                for s in TrvObservationSummary.objects.filter(
+                    observation_id__in=obs_ids
```
Comment (RickWangPerth) [inline wamtram2/views.py:2457]: **P1: The summary-view lookup also exceeds SQL Server's parameter limit for large exports.**

This query sends the entire `obs_ids` set in one `IN (...)` clause and is not protected by the helper's error handling, so a large export fails with a 500 response. Please fetch summary rows in the same safe-sized chunks as the other related data.

File: wamtram2/export_helpers.py [hunk 5]
```diff
@@ -0,0 +1,1173 @@
...
+    if isinstance(value, date):
+        return value.strftime("%m/%d/%Y")
+
+    if isinstance(value, bool):
+        return "True" if value else "False"
+
+    if value is None:
+        return ""
+
+    return str(value)
```
Comment (RickWangPerth) [inline wamtram2/export_helpers.py:119]: **P2: Escape formula-leading user text before writing CSV/XLSX cells.**

Free-text fields such as comments are passed through unchanged into both formats. A value starting with `=`, `+`, `-`, or `@` can be interpreted as a spreadsheet formula when the recipient opens the export. Please sanitize formula-leading text or explicitly write it as text.

File: wamtram2/views.py (hunk 4 above)
Comment (LizzzzHFFF) [reply wamtram2/views.py:2457]: same  as T002

File: wamtram2/export_helpers.py (hunk 5 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py:119]: same as T002

File: wamtram2/export_helpers.py (hunk 3 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py:386]: same as T002

File: wamtram2/export_helpers.py (hunk 2 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py:511]: same as T002

File: wamtram2/export_helpers.py (hunk 1 above)
Comment (LizzzzHFFF) [reply wamtram2/export_helpers.py:612]: resolved in T070
